### PR TITLE
Updated lab 2 for 2019

### PR DESCRIPTION
**Pull request by Tclv**

### CODE REVIEW

**Comment by Tclv** [comment]
@mbrobbel 🦇 💡

File: doc/2-guide.text [hunk 1]
```diff
@@ -1039,6 +1039,158 @@ Be sure to explore the history server thoroughly! You can use it to gain an
...
+To move a file
+
+``` {.bash}
+aws s3 mv path/to/file s3://destination-bucket/path/to/file
+```
+
+The aws-cli contains much more functionality, which can be found on the
+[AWS-CLI docs].
+
+Once you have uploaded all the necessary files (again your application JAR, and
```
Comment (mbrobbel) [inline doc/2-guide.text]: This sentence seems to miss part 2.

File: doc/2-guide.text [hunk 2]
```diff
@@ -1039,6 +1039,158 @@ Be sure to explore the history server thoroughly! You can use it to gain an
...
+    your account is allowed to provision. If you don't have access to enough
+    spot instances, the procedure to request additional can be found in the
+    [AWS documentation].
+
+In the *General Options* you can select a cluster name. You can tune where the
+system logs and a number of other features (more information in the popups).
+After finishing this step, press *next*.
+
+You should now arrive in the *Security Options* screen. If you have not created
+a *EC2 keypair*, it is recommended that you do so now. This will allow you to
```
Comment (mbrobbel) [inline doc/2-guide.text]: **an** EC2 keypair

File: doc/2-guide.text [hunk 3]
```diff
@@ -1039,6 +1039,158 @@ Be sure to explore the history server thoroughly! You can use it to gain an
...
+access the Yarn, Spark, and Ganglia web interfaces in your browser. This makes
+debugging and monitoring the execution of your Spark Job much more manageable.
+To create a *EC2 keypair*, follow [these instructions][keypair instructions].
```
Comment (mbrobbel) [inline doc/2-guide.text]: an

File: doc/2-guide.text [hunk 4]
```diff
@@ -1039,6 +1039,158 @@ Be sure to explore the history server thoroughly! You can use it to gain an
...
+
+After this has all been completed you are ready to spin up your first cluster
+by pressing *Create cluster*. Once the cluster has been created, AWS will start
+provisioning machines. This should take about 10 minutes. In the meantime you
+can add a step. Go the *Steps* foldout, and select *Spark application* for
```
Comment (mbrobbel) [inline doc/2-guide.text]: Go **to** the

File: doc/2-guide.text [hunk 5]
```diff
@@ -1039,6 +1039,158 @@ Be sure to explore the history server thoroughly! You can use it to gain an
...
+*Step Type*. Clicking on *Configure* will open a dialogue in which you can
+select the application JAR location in your S3 bucket, as well as any number
+of argument to the application, spark-submit, as well as your action on
```
Comment (mbrobbel) [inline doc/2-guide.text]: argument**s**

File: doc/4-lab2.text [hunk 6]
```diff
@@ -205,24 +56,16 @@ There is a [guide to Spark performance] tuning on the Spark website.
 
 ## Deliverables
 
--   A report outlining your choices in terms of configuration and your results.
--   A presentation (maximum 5 slides/minutes) in which you present your work
-    and results to the class. Try to put an emphasis on the
-    improvements you found, what kind of settings/configurations/changes had
-    the most impact.
+The deliverable for this lab will be a blog post outlining your choices in
```
Comment (mbrobbel) [inline doc/4-lab2.text]: will be -> is

**Pull request merged**